wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -29,9 +31,47 @@
  *
  * @return string The current URL.
  */
-function get_current_url() {
+function get_filtered_url() {
 	global $wp;
-	return home_url( add_query_arg( array(), $wp->request ) );
+
+	$current_url = home_url( add_query_arg( array(), $wp->request ) );
+
+	// Remove the page query var.
+	// If the path retains "page," the filtered result might be a 404 page.
+	$parsed_url = wp_parse_url( $current_url );
+	$path = isset( $parsed_url['path'] ) ? $parsed_url['path'] : '';
+	$filtered_path = preg_replace( '#/page/\d+/?$#', '', $path );
+
+	return home_url() . $filtered_path;
+}
+
+/**
+ * Get the content type options.
+ * Used for the search filters and the archive filters.
+ *
+ * @param array $options The options for this filter.
+ * @return array New list of custom content type options.
+ */
+function get_content_type_options( $options ) {
+	global $wp_query;
+
+	$options = array(
+		'any' => __( 'Any', 'wporg-learn' ),
+		'course' => __( 'Course', 'wporg-learn' ),
+		'lesson' => __( 'Lesson', 'wporg-learn' ),
+	);
+
+	$selected_slug = $wp_query->get( 'post_type' ) ? $wp_query->get( 'post_type' ) : 'any';
+	$label = get_label_by_slug( $selected_slug, $options );
```

Comment:
Since these are keyed by the slug, do you need an extra function? I think this should work for both here and `get_student_course_options` (though the fallback there should be `all`).

```suggestion
	$label = $options[ $selected_slug ] ?? $options['any'];
```

Reply:
yeah, you're exactly right, those logic are unnecessary..
I removed the `?? $options['any'];` as `$options[ $selected_slug ]` will always have a value.
18c28a6

Reply:
It will, but it could be something not in the list — if someone manually types in a post type, for example. Adding the fallback works as a guard, just in case.

Reply:
ah, yes. Missed the fact that it fetches the value from the URL. ae09e8e